File: ruby/example_code/workdocs/wd_list_users.rb
# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
# SPDX - License - Identifier: Apache - 2.0

# Purpose
# This code example demonstrates how to describe your Amazon WorkDocs users.

# snippet-start:[s3.wd_list_users.rb]

require 'aws-sdk-workdocs'  # v2: require 'aws-sdk'
# Replace us-west-2 with the AWS Region you're using for Amazon WorkDocs.
client = Aws::WorkDocs::Client.new(region: 'us-west-2')

# Set to the OrganizationId of your WorkDocs site
orgId = 'd-123456789c'

resp = client.describe_users({
  organization_id: orgId,
  include: "ALL", # accepts ALL, ACTIVE_PENDING
  order: "ASCENDING", # accepts ASCENDING, DESCENDING
  sort: "USER_NAME", # accepts USER_NAME, FULL_NAME, STORAGE_LIMIT, USER_STATUS, STORAGE_USED
})

resp.users.each do |user|
  puts "First name:  #{user.given_name}"
  puts "Last name:   #{user.surname}"
  puts "Email:       #{user.email_address}"
  puts "Root folder: #{user.root_folder_id}"
  puts
end
# snippet-end:[s3.wd_list_users.rb]

File: go/example_code/workdocs/wd_list_users.go
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0
// snippet-start:[workdocs.go.list_users.complete]
package main

// snippet-start:[workdocs.go.list_users.imports]
import (
	"os"

	"github.com/aws/aws-sdk-go/aws/session"
	"github.com/aws/aws-sdk-go/service/workdocs"

	"flag"
	"fmt"
)

// snippet-end:[workdocs.go.list_users.imports]

/*
  Lists all user names or extra info about user USER_NAME

  Usage:
    go run wd_list_users.go [USER_NAME]
*/

func main() {
	// snippet-start:[workdocs.go.list_users.vars]
	orgPtr := flag.String("o", "", "The ID of your organization")
	userPtr := flag.String("u", "", "User for whom info is retrieved")

	flag.Parse()

	if *orgPtr == "" {
		fmt.Println("You must supply the organization ID")
		flag.PrintDefaults()
		os.Exit(1)
	}
	// snippet-end:[workdocs.go.list_users.vars]

	// snippet-start:[workdocs.go.list_users.input]
	input := new(workdocs.DescribeUsersInput)
	input.OrganizationId = orgPtr

	// Show all users if we don't get a user name
	if *userPtr == "" {
		fmt.Println("Getting info about all users")
	} else {
		fmt.Println("Getting info about user " + *userPtr)
		input.Query = userPtr
	}
	// snippet-end:[workdocs.go.list_users.input]

	// Initialize a session that the SDK will use to load
	// credentials from the shared credentials file. (~/.aws/credentials).
	// snippet-start:[workdocs.go.list_users.session]
	sess := session.Must(session.NewSessionWithOptions(session.Options{
		SharedConfigState: session.SharedConfigEnable,
	}))

	svc := workdocs.New(sess)
	// snippet-end:[workdocs.go.list_users.session]

	fmt.Println("")

	// snippet-start:[workdocs.go.list_users.describe]
	result, err := svc.DescribeUsers(input)
	if err != nil {
		fmt.Println("Error getting user info", err)
		return
	}

	if *userPtr == "" {
		fmt.Println("Found", *result.TotalNumberOfUsers, "users")
		fmt.Println("")
	}

	for _, user := range result.Users {
		fmt.Println("Username:   " + *user.Username)

		if *userPtr != "" {
			fmt.Println("Firstname:  " + *user.GivenName)
			fmt.Println("Lastname:   " + *user.Surname)
			fmt.Println("Email:      " + *user.EmailAddress)
			fmt.Println("Root folder " + *user.RootFolderId)
		}

		fmt.Println("")
	}
	// snippet-end:[workdocs.go.list_users.describe]
}

// snippet-end:[workdocs.go.list_users.complete]
